entity-framework/core/querying/complex-query-operators.md

```diff
@@ -0,0 +1,128 @@
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#Join)]
+
+```SQL
+SELECT [p].[PersonId], [p].[Name], [p].[PhotoId], [p0].[PersonPhotoId], [p0].[Caption], [p0].[Photo]
+FROM [PersonPhoto] AS [p0]
+INNER JOIN [Person] AS [p] ON [p0].[PersonPhotoId] = [p].[PhotoId]
+```
+
+## GroupJoin
+
+LINQ GroupJoin operator allows you to connect two data sources in similar to Join but it creates a group of inner values for matching outer elements. Executing query like following generates result of `Blog` & `IEnumerable<Post>`. Since databases (especially relational databases) don't support representing a collection of client-side objects, GroupJoin can't be translated to server in many cases. It requires to get all the data from server to do GroupJoin without special selector (first query below). But if the selector is limiting data being selected then fetching all data from server may cause performance issues (second query below). That's why EF Core doesn't translate GroupJoin.
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#GroupJoin)]
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#GroupJoinComposed)]
+
+## SelectMany
+
+LINQ SelectMany operator allows you to enumerate over a collection selector for each outer element and generate tuples of value from each data source. In a way, it's a join but without any condition so every outer element is connected with element from collection source. Depending on how the collection selector is related to the outer data source, SelectMany can translate to various different queries on the server side.
+
+### Collection selector doesn't reference outer
+
+When the collection selector isn't referencing anything from the outer source, the result is a cartesian product of both data sources. It translates to `CROSS JOIN` in relational databases.
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#SelectManyConvertedToCrossJoin)]
+
+```SQL
+SELECT [b].[BlogId], [b].[OwnerId], [b].[Rating], [b].[Url], [p].[PostId], [p].[AuthorId], [p].[BlogId], [p].[Content], [p].[Rating], [p].[Title]
+FROM [Blogs] AS [b]
+CROSS JOIN [Posts] AS [p]
+```
+
+### Collection selector references outer in where clause
+
+When the collection selector has a where clause, which references the outer element, then EF Core translates it to a database join and uses the predicate as the join condition. Normally it arises when using collection navigation on the outer element as the collection selector. If the collection is empty for outer element, then no results would be generated for that outer element. But if `DefaultIfEmpty` is applied on the collection selector then the outer element will be connected with default value of the inner element. Because of this distinction, this kind of queries translates to `INNER JOIN` in the absence of `DefaultIfEmpty` and `LEFT JOIN` when `DefaultIfEmpty` is applied.
```

Comment:
```suggestion
When the collection selector has a where clause, which references the outer element, then EF Core translates it to a database join and uses the predicate as the join condition. Normally this arises when using collection navigation on the outer element as the collection selector. If the collection is empty for an outer element, then no results would be generated for that outer element. But if `DefaultIfEmpty` is applied on the collection selector then the outer element will be connected with a default value of the inner element. Because of this distinction, this kind of queries translates to `INNER JOIN` in the absence of `DefaultIfEmpty` and `LEFT JOIN` when `DefaultIfEmpty` is applied.
```